File: services/backend/doc-server/src/main/java/com/meta/doc/dtos/DocsDTO.java
package com.meta.doc.dtos;

import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class DocsDTO {
    private String id;
    private String teamId;
    private String officeId;
    private String parentId;
    private String title;
    private String content;
    private String rootGrandparentId;
    private List<DocsDTO> children;
    private int level;
    private List<DocumentFileDTO> files;
}

File: apps/frontend/src/types/DocsDTO.ts
// src/types/DocsDTO.ts

export interface DocsDTO {
  id: string;
  teamId: string;
  officeId: string;
  parentId: string | null;
  title: string;
  content: string;
  rootGrandparentId: string | null;
  children?: DocsDTO[];
  level: number;
}
